### ISSUE TITLE
MD044: YAML keys are treated as prose

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#448]
### rumdl version

0.1.25

### Installation method

mise

### Bug description

Rule MD044 (https://github.com/rvben/rumdl/blob/ee659eacb863b89d32cb8c64833e9af3c56daa61/docs/md044.md) can check for capitalisation of prose. This test extends to YAML frontmatter that is present in many Markdown files.

It's fairly customary for YAML keys to be all lowercase but, if a key happens to match `names` configuration in MD044, the key will be flagged as a violation of the rule. This is unlikely to be wanted.

### Steps to reproduce

1. Create a Markdown file with the following content:
   ```markdown
   ---
   title: Heading
   test: Some Test value
   ---

   Test
   ```
1. Create a `.rumdl.toml` file with the following content:
   ```toml
   [MD044]
   names = ["Test"]
   ```
1. Run `rumdl check` on the Markdown file and see that `test` is flagged.

### Expected behavior

I'd expect no YAML key (only values for those keys) to be flagged by an entry in `names` for MD044 because they're not prose and are usually all lowercase.

### Actual behavior

`test` doesn't match the required capitalisation `Test` in the `names` configuration and so `rumdl` flags a violation.

### Configuration

```toml
[MD044]
names = ["Test"]
```

### Operating system

macOS

**Comment by eread** [comment]
Thanks for fixing this @rvben. Version `0.1.26` ignores keys when applying rule MD044.

It also looks like it ignores the values. So I'd expect, using the configuration above, one violation in this Markdown:

```markdown
---
title: Heading
test: Some test value
---

Test
```

Here, _Some test value_ violates MD044 because it is prose. I think as long as a YAML _value_ is a string, it could be considered prose. WDYT?

**Comment by rvben** [comment]
Good catch on the follow-up @eread. Fixed in v0.1.27. YAML values are now checked as prose while keys, delimiters, comments, and flow constructs (`{}`, `[]`) are skipped. TOML keys handled correctly too.

**Comment by eread** [comment]
> Good catch on the follow-up [@eread](https://github.com/eread). Fixed in v0.1.27. YAML values are now checked as prose while keys, delimiters, comments, and flow constructs (`{}`, `[]`) are skipped. TOML keys handled correctly too.

@rvben Thanks!
